- Description: multicast ping doesn't work
  Symptom:     issueing a multicast ping generates no multicast traffic,
               means no packets are received by other systems
  Problem:     command structure for registering multicast addresses was misaligned
  Solution:    pack the structure so there are no holes anymore
  Problem-ID:  -
  Archs:       s390-31, s390-64

--- linux-2.5/drivers/s390/net/lcs.h	11 May 2004 08:59:46 -0000	1.15.2.1
+++ linux-2.5/drivers/s390/net/lcs.h	17 Jun 2004 08:11:11 -0000	1.15.2.2
@@ -6,7 +6,7 @@
 #include <linux/workqueue.h>
 #include <asm/ccwdev.h>
 
-#define VERSION_LCS_H "$Revision: 1.15.2.1 $"
+#define VERSION_LCS_H "$Revision: 1.15.2.2 $"
 
 #define LCS_DBF_TEXT(level, name, text) \
 	do { \
@@ -221,8 +221,8 @@
 				struct lcs_ip_mac_pair
 				ip_mac_pair[32];
 				__u32	  response_data;
-			} lcs_ipass_ctlmsg;
-		} lcs_qipassist;
+			} lcs_ipass_ctlmsg __attribute__ ((packed));
+		} lcs_qipassist __attribute__ ((packed));;
 #endif /*CONFIG_IP_MULTICAST */
 	} cmd __attribute__ ((packed));
 }  __attribute__ ((packed));

